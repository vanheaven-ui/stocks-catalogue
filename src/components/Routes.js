import { BrowserRouter as Router, Route, Switch } from 'react-router-dom';
import Navbar from '../common/components/Navbar';
import Details from './pages/Details';
import Home from './pages/Home';

const Routes = () => (
  <Router>
    <Navbar />
    <div className="App">
      Starting out!
      <Switch>
        <Route exact path="/">
          <Home />
        </Route>
        <Route path="/stocks/:id">
          <Details />
        </Route>
      </Switch>
    </div>
  </Router>
);

export default Routes;
