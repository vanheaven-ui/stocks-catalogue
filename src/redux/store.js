import { compose, createStore } from 'redux';
import rootReducer from './reducers';

const storeEnhancers = compose(
  window.__REDUX_DEVTOOLS_EXTENSION__ && window.__REDUX_DEVTOOLS_EXTENSION__(), // eslint-disable-line
);

const store = createStore(rootReducer, storeEnhancers);

export default store;
