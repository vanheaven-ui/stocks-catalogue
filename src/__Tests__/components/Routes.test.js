import ReactDOM from 'react-dom';
import { Provider } from 'react-redux';
import renderer from 'react-test-renderer';
import { cleanup } from '@testing-library/react'; // render
import { BrowserRouter } from 'react-router-dom';
import Routes from '../../components/Routes';
import rootNode from '../helpers';
import store from '../../redux/store';

describe('Routes component', () => {
  afterEach(cleanup);

  test('renders without crashing', () => {
    ReactDOM.render(
      <BrowserRouter><Provider store={store}><Routes /></Provider></BrowserRouter>, rootNode(),
    );
  });

  test('doesnot change unexpectedly', () => {
    const tree = renderer.create(
      <BrowserRouter><Provider store={store}><Routes /></Provider></BrowserRouter>,
    ).toJSON();
    expect(tree).toMatchSnapshot();
  });
});
