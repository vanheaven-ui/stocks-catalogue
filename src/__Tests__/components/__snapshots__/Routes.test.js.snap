// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Routes component doesnot change unexpectedly 1`] = `
Array [
  <nav
    className="navbar sticky-top mb-2"
    style={
      Object {
        "color": "#007bff",
      }
    }
  >
    <h2>
      <a
        href="/"
        onClick={[Function]}
      >
        <img
          alt="my logo"
          src="logo.jpeg"
          style={
            Object {
              "height": 60,
              "marginRight": 5,
              "width": 60,
            }
          }
        />
        StocksCatalogue
      </a>
    </h2>
    <div
      className="links"
    >
      <a
        href="/"
        onClick={[Function]}
        style={
          Object {
            "margin": "0 10px 0 0",
          }
        }
      >
        Home
      </a>
      <a
        href="/news"
        onClick={[Function]}
      >
        StockNews
      </a>
    </div>
  </nav>,
  <div
    className="App"
  >
    <section
      className="home"
    >
      <section
        className="hero"
      >
        <div
          className="container-fluid"
        >
          <div
            className="row"
            style={
              Object {
                "height": 362,
              }
            }
          >
            <div
              className="col-lg-7 col-sm-12"
              style={
                Object {
                  "height": "inherit",
                }
              }
            >
              <div
                className="bg-dark text-white card"
                style={
                  Object {
                    "height": 200,
                  }
                }
              >
                <img
                  alt="Card image"
                  className="card-img"
                  src="hero.jpg"
                />
                <div
                  className="card-img-overlay"
                >
                  <div
                    className="card-title h5"
                    style={
                      Object {
                        "fontSize": 48,
                        "textShadow": "3px 2px black",
                      }
                    }
                  >
                    INFORMATION FOR YOUR
                    <br />
                    NEXT STOCKS INVESTMENT
                  </div>
                </div>
              </div>
            </div>
            <div
              className="col-lg-5 col-sm-12"
              style={
                Object {
                  "height": "inherit",
                }
              }
            >
              <div
                className="bg-dark text-light card border-none"
                style={
                  Object {
                    "height": "inherit",
                    "verticleAlign": "middle",
                  }
                }
              >
                <img
                  alt="Card image"
                  className="card-img"
                  src="hero-bar.png"
                  style={
                    Object {
                      "margin": "0 auto",
                      "width": 300,
                    }
                  }
                />
                <div
                  className="card-img-overlay"
                >
                  <div
                    className="card-title h5"
                  >
                    <div
                      className="fade alert alert-primary alert-dismissible show"
                      role="alert"
                    >
                      <button
                        className="close"
                        onClick={[Function]}
                        type="button"
                      >
                        <span
                          aria-hidden="true"
                        >
                          Ã—
                        </span>
                        <span
                          className="sr-only"
                        >
                          Close alert
                        </span>
                      </button>
                      <div
                        className="alert-heading h4"
                        id="heading"
                      >
                        <span
                          aria-labelledby="heading"
                          role="img"
                        >
                          ðŸ‘‹
                        </span>
                        Hey!, Welcome to stocks catalogue
                      </div>
                      <hr />
                      <p
                        style={
                          Object {
                            "fontSize": 16,
                          }
                        }
                      >
                        This is a Single Page Application(SPA) that consumes the
                         
                        <a
                          href="https://financialmodelingprep.com/developer/docs/"
                          rel="noreferrer"
                          style={
                            Object {
                              "fontWeight": 700,
                              "textDecoration": "underline",
                            }
                          }
                          target="_blank"
                        >
                          financialmodelingprep API
                           
                        </a>
                        to get real time stock trading details and news so that they are easily accessible.
                        <br />
                        <strong>
                          Remember to use the filter to narrow down your browsing list.
                        </strong>
                        <br />
                        Enjoy and make the best decision in this volatile business!
                      </p>
                    </div>
                  </div>
                </div>
              </div>
            </div>
          </div>
        </div>
      </section>
      <section
        className="stock-list"
      >
        <div
          className="container-fluid"
          style={
            Object {
              "backgroundColor": "#00800080",
              "padding": 0,
            }
          }
        >
          <section
            className="filters"
            style={
              Object {
                "backgroundColor": "#ffcd0a",
                "color": "#d1450d",
              }
            }
          >
            <h2>
              Browse Stocks with Filters:
            </h2>
            <div
              className="container"
            >
              <form
                onSubmit={[Function]}
                style={
                  Object {
                    "textAlign": "center",
                  }
                }
              >
                <div
                  className="filters"
                >
                  <select
                    name="PRICE"
                    onChange={[Function]}
                    value="All"
                  >
                    <option
                      value="default"
                    >
                      Select:
                       
                      PRICE RANGE
                    </option>
                    <option
                      value="<100"
                    >
                      &lt;100
                    </option>
                    <option
                      value="100 < Price < 200"
                    >
                      100 &lt; Price &lt; 200
                    </option>
                    <option
                      value="200 < Price < 300"
                    >
                      200 &lt; Price &lt; 300
                    </option>
                    <option
                      value="300 < Price < 400"
                    >
                      300 &lt; Price &lt; 400
                    </option>
                    <option
                      value="400 < Price < 500"
                    >
                      400 &lt; Price &lt; 500
                    </option>
                    <option
                      value=">500"
                    >
                      &gt;500
                    </option>
                  </select>
                  <select
                    name="EXCHANGE"
                    onChange={[Function]}
                    value="All"
                  >
                    <option
                      value="default"
                    >
                      Select:
                       
                      EXCHANGE
                    </option>
                  </select>
                  <button
                    type="submit"
                  >
                    Apply
                  </button>
                </div>
              </form>
            </div>
          </section>
          <div
            className="px-2 row"
          />
          <ul
            className="myClass"
          >
            <li
              className="previous disabled"
            >
              <a
                aria-disabled="true"
                aria-label="Previous page"
                className="prev"
                onClick={[Function]}
                onKeyPress={[Function]}
                rel="prev"
                role="button"
                tabIndex="0"
              >
                PREV
              </a>
            </li>
            <li
              className="next"
            >
              <a
                aria-disabled="false"
                aria-label="Next page"
                className="next"
                onClick={[Function]}
                onKeyPress={[Function]}
                rel="next"
                role="button"
                tabIndex="0"
              >
                NEXT
              </a>
            </li>
          </ul>
        </div>
      </section>
    </section>
  </div>,
]
`;
